Cauldron/Assets/Scripts/Wardrobe/Wardrobe.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using DG.Tweening;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+using Zenject;
+namespace CauldronCodebase
+{
+    public class Wardrobe : Book
+    {
+        [SerializeField] private RectTransform descriptionPanel;
+        [SerializeField] private TMP_Text description;
+        
+        [SerializeField] private Button confirmSkinButton;
+        [SerializeField] private Button closeButton;
+        
+        [SerializeField] private SkinSO[] skinModels;
+        [SerializeField] private WardrobeCell[] wardrobeCells;
+
+        [Header("Animation")]
+        [SerializeField] private float descriptionPanelShowSpeed;
+        [SerializeField] private float descriptionPanelXPos;
+
+        [Inject] private WitchSkinChanger witchSkinChanger;
+
+        private WardrobeCell _selectedCell;
+        private float _initDescriptionPanelPos;
+
+        private void Start()
+        {
+            closeButton.onClick.AddListener(CloseBook);
+            confirmSkinButton.onClick.AddListener(CloseBook);
+            _initDescriptionPanelPos = descriptionPanel.anchoredPosition.x;
+        }
+
+        protected override void InitTotalPages()
+        {
+            totalPages = 1;
+        }
+        
+        protected override void UpdatePage()
+        {
+            var unlockedSkins = GetUnlockedSkins();
+            for (int i = 0; i < wardrobeCells.Length; i++)
+            {
+                skinModels[i].IsAvailable = unlockedSkins.Contains(skinModels[i].SkinName);
```

Comment:
We need a skinProvider that will remember all the skins user has bought/acquired, because they are not tied directly to endings now. It can contain the links to all skin SOs, so that they are all in one place. Endings can unlock skins on their own, so they should have an optional skinSO field.